### PR TITLE
Remove lock files and test github masters in Kitchen Tests

### PR DESCRIPTION
**Pull request by tas50**
Remove deps we don't actually use
Pull things we control from master
Remove the Berkshelf and Gem lock files so we test the latest things available. Otherwise we're testing things from months ago, which isn't what customers would use or what we ship in ChefDK.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: kitchen-tests/Gemfile [hunk 1]
```diff
@@ -1,13 +1,9 @@
 source "https://rubygems.org"
 
-gem "chef-config", "< 13"
 gem "rake" # required to build some native extensions
 gem "berkshelf"
```
Comment (lamont-granquist) [inline kitchen-tests/Gemfile]: you could try pulling berks from master here

Comment (tas50) [reply kitchen-tests/Gemfile]: Do we need to do that or *should* we just to test that

Comment (lamont-granquist) [reply kitchen-tests/Gemfile]: its a _should_ not a _need_, although technically i think we should nuke the Gemfile.lock and pull from git master on everything we control (unless there's some issue requiring pinning)

**Comment by tas50** [comment]
@lamont-granquist this work for you now?

**Pull request merged**